### PR TITLE
docs(tile, tile-group): deprecate components

### PR DESCRIPTION
**Pull request by DitwanP**
**Related Issue:** #9253 

## Summary
Adds a JSDoc `@deprecated` tag to the following components:
- `tile-select`
- `tile-select-group`

### CODE REVIEW

**Review by geospatialem — COMMENTED** [review]
Awesome! We'll need to tag team on the naming convention, as https://github.com/Esri/calcite-design-system/pull/9285 uses `calcite-` and this is PR currently scoped to the component name.

WDYT is the best approach?

**Comment by DitwanP** [comment]
> Awesome! We'll need to tag team on the naming convention, as #9285 uses `calcite-` and this is PR currently scoped to the component name.
> 
> WDYT is the best approach?

I don't think it makes a huge difference but if we are to make a decision going forward I prefer the "calcite-..." option.

File: packages/calcite-components/src/components/tile-select-group/tile-select-group.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ import {
 import { TileSelectGroupLayout } from "./interfaces";
 
 /**
+ * @deprecated  Use the `calcite-tile-group` component instead.
```
Comment (geospatialem) [inline packages/calcite-components/src/components/tile-select-group/tile-select-group.tsx:12]: nitpick:

It looks like there's two spaces between the `@deprecated` and `"Use"`.

File: packages/calcite-components/src/components/tile-select/tile-select.tsx [hunk 2]
```diff
@@ -30,6 +30,7 @@ import { TileSelectType } from "./interfaces";
 import { CSS } from "./resources";
 
 /**
+ * @deprecated  Use the `calcite-tile` component instead.
```
Comment (geospatialem) [inline packages/calcite-components/src/components/tile-select/tile-select.tsx:33]: nitpick:

It looks like there's two spaces between the `@deprecated` and `"Use"`.

**Review by geospatialem — APPROVED** [review]
**_Awesome!_** 🌟📚🌟

File: packages/calcite-components/src/components/tile-select/tile-select.tsx (hunk 2 above)
Comment (DitwanP) [reply packages/calcite-components/src/components/tile-select/tile-select.tsx:33]: 🦅👁️ Good catch!

**Review by geospatialem — APPROVED** [review]

**Pull request merged**